api/server.py

```diff
@@ -281,8 +281,35 @@ def decorated(*args, **kwargs):
 def healthcheck():
     """No access token required to access this route"""
     now = datetime.datetime.now()
-    response = "CR3 Download API - Health Check - Available @ %s" % now.strftime(
-        "%Y-%m-%d %H:%M:%S"
+
+    # Read the system uptime from /proc/uptime
+    with open("/proc/uptime", "r") as f:
+        uptime_seconds = float(f.readline().split()[0])
+        uptime_str = str(datetime.timedelta(seconds=uptime_seconds)).split(".")[
+            0
+        ]  # Remove microseconds for a terse format
+
+    # Get the process start time
+    pid = os.getpid()
+    with open(f"/proc/{pid}/stat", "r") as f:
+        proc_start_time_ticks = int(f.readline().split()[21])
+        proc_start_time_seconds = proc_start_time_ticks / os.sysconf(
+            os.sysconf_names["SC_CLK_TCK"]
+        )
+        proc_uptime_seconds = uptime_seconds - proc_start_time_seconds
+        proc_uptime_str = str(datetime.timedelta(seconds=proc_uptime_seconds)).split(
+            "."
+        )[
+            0
+        ]  # Remove microseconds for a terse format
+
+    response = (
+        "CR3 Download API - Health Check - Available @ %s - System Uptime: %s - Process Uptime: %s"
+        % (
+            now.strftime("%Y-%m-%d %H:%M:%S"),
+            uptime_str,
+            proc_uptime_str,
+        )
```

Comment:
cool—i had no idea how one would go about showing "uptime". do we have any concern about exposing this method publicly? would it be less scary if the health check just printed the current time and said everything's ok? i guess i'm just double-checking that exposing system calls on a public route is fine 😅

Reply:
I don't think it's any harm in showing the uptime, but I'm also happy to take it out. It was very handing knowing when a new version of the docker container had been fired up, so I thought it might want to stay around.

There is not time when we believe our service to go from secure to insecure, so exposing the length of time doesn't seem to have any downside, but again, no harm in taking it out either.

Reply:
> when a new version of the docker container had been fired up,

Mike brought up a good point in the skillshare, if theres a way to include what version is running. Future enhancement for sure